### PR TITLE
Use pos.unit.source instead of currentUnit.source in quick fixes

### PR DESCRIPTION
**Pull request by lrytz**
Type completion can lead to currentUnit being different from tree.pos.unit.

Test case for https://github.com/scala/bug/issues/12941, fix was in https://github.com/scala/scala/pull/10681.

### CODE REVIEW

File: test/junit/scala/tools/nsc/QuickfixTest.scala [hunk 1]
```diff
@@ -6,25 +6,30 @@ import org.junit.Test
 import java.nio.file.Files
 import scala.reflect.internal.util.BatchSourceFile
 import scala.reflect.io.AbstractFile
+import scala.tools.nsc.reporters.StoreReporter
 import scala.tools.testkit.BytecodeTesting
 import scala.tools.testkit.ReleasablePath._
 import scala.util.Using
 
-import reporters.StoreReporter
```
Comment (som-snytt) [inline test/junit/scala/tools/nsc/QuickfixTest.scala:13]: I thought of you when I typed that at midnight, knowing you'd want me to use more keys on the keyboard.

File: test/junit/scala/tools/nsc/QuickfixTest.scala [hunk 2]
```diff
@@ -6,25 +6,30 @@ import org.junit.Test
...
-      Files.write(src, a.getBytes)
-      val c = BytecodeTesting.newCompiler(extraArgs = args)
-      val r = c.newRun()
-      val f = AbstractFile.getFile(src.toFile.getAbsolutePath)
-      r.compileSources(List(new BatchSourceFile(f)))
-      assertEquals(b, new String(Files.readAllBytes(src)))
-      for (info <- c.global.reporter.asInstanceOf[StoreReporter].infos)
-        assert(checkInfo(info), info)
+
+  def testQuickfixs(as: List[String], b: String, args: String, checkInfo: StoreReporter.Info => Boolean = _ => true): Unit =
```
Comment (som-snytt) [inline test/junit/scala/tools/nsc/QuickfixTest.scala:16]: I didn't want to provoke you by indenting the `if`, but glad you did that and added multiple sources.

**Review by som-snytt — APPROVED** [review]
Thanks, I guessed there must be more usages to fix.

File: test/junit/scala/tools/nsc/QuickfixTest.scala [hunk 3]
```diff
@@ -6,25 +6,30 @@ import org.junit.Test
...
+    if (!scala.util.Properties.isWin) {
+      Using.resource(Files.createTempDirectory("quickfixTest")) { tmpDir =>
+        val srcs = as.indices.map(i => tmpDir.resolve(s"unitSource$i.scala")).toList
+        as.lazyZip(srcs).foreach { case (a, src) => Files.write(src, a.getBytes) }
+        val c = BytecodeTesting.newCompiler(extraArgs = args)
+        val r = c.newRun()
+        val fs = srcs.map(src => AbstractFile.getFile(src.toFile.getAbsolutePath))
+        r.compileSources(fs.map(new BatchSourceFile(_)))
+        assertEquals(b.replaceAll("\n+", "\n"), srcs.map(src => new String(Files.readAllBytes(src))).mkString("\n").replaceAll("\n+", "\n"))
```
Comment (som-snytt) [inline test/junit/scala/tools/nsc/QuickfixTest.scala:25]: Not sure why merged result was easier, but by coincidence, someone noted on discord that `\R\R` and `\R{2}` behave differently on windows `\r\n`, and I tried to think of any instance where I needed that. In general, it's better to just not run on windows, problem solved.

**Comment by som-snytt** [comment]
Just for fun, squash-merged since it is a single commit.

**Comment by SethTisue** [comment]
Daring!

We'll see whether Lukas lets you get away with it. No-squash-merging is an old Adriaan policy we haven't revisited in recent years. Though if it remains the policy, why aren't squash merges forbidden in the GitHub settings? 🤔 

Anyway, I don't have a strong feeling about it. And the policy predates me joining the team but Lukas might remember what the considerations were.

**Comment by som-snytt** [comment]
As a contributor, it means I must rebase my branch so `git branch -d` works. Chris did this to me once & opened my eyes. It does make the history look nicer, especially for one measly commit. (Not to minimize Lukas's contribution here.)

Someone (lrytz) has been helpfully updating my branches anyway, so I'm more accustomed to waking up to different commits. Sometimes you wake up next to someone and wonder if you're really committed.

Edit: really I just wanted to try out the button, rather than change policy.

File: test/junit/scala/tools/nsc/QuickfixTest.scala (hunk 1 above)
Comment (lrytz) [reply test/junit/scala/tools/nsc/QuickfixTest.scala:13]: I like IntelliJ's Organize Imports myself. My opinions about imports don't go any deeper than that.

**Comment by lrytz** [comment]
I don't remember the reason from the top of my head. There's some value in the uniformity, like for searching history. So I'd keep using merges.

**Comment by SethTisue** [comment]
Okay, I've disabled squash merging at https://github.com/scala/scala/settings so we don't do it by accident.

**Pull request merged**